Dashboard integration contact group assoc

aurelienCastellarnau opened the pull request:
Interfaces: 
- PenelopeController => permet une initialisation plus simple des controllers
- ContactDAOReceipe => conditionne le contenu de ContactDAO
Ajouts:
- AppController => se charge d'instancier les controllers et les vues dans le bon ordre
- Contact workflow => chargement de GroupPanels dans le ContactPanel
- Liens entre le nouveau dashboard tout beau et le refresh du model
Les actions du crud sur Contact sont toutes fonctionnelles et provoquent un refresh.
Déroulement de l'archi:
1 - AppController appelle init(DataInterface) sur les controllers enfants
    => init instancie le DAO correspondant par controller et enregistre le controller comme listener du DAO
2 - AppController instancie la main view en lui passant les controllers.
3 - La main view est stockée dans chaque controller.
Durant le runtime:
1 - chaque controller écoute sa partie de vue et son dao (MVC)
2 - les actions 'pure-vue' (afficher les contacts, afficher les groups et peut être à terme, déclencher un update...) sont gérées dans le package ihm par le biais d'un pattern observer  (ca n'a pas bougé)

Commit:
Controller initialisation:
Comme le Dashboard n'est pas un controller j'ai voulu donner un moyen de gérer les PenelopeCotnroller
comme un ensemble lors de l'initialisation mais aussi individuellement pour qu'on puisse les manipuler
dans le dashboard et les enregistrer un par un dans les observer correspondant.
On stock donc les controllers dans une HashMap, ca permet de les retrouver par clé.
Du coup les vues ne possèdent plus de DAO, c'est le controller chargé de gérer cette partie qui fournis le sien.

Commit:
Model: ajout d'un doublon dans les propriétés qui gèrent les ids.
Pour nous c'est transparent: un addGroup ou un setGroups va automatiquement ajouté les id qu'il faut dans l'objet.
Niveau sérialisation, les champs id sont écris dans le fichier là ou les champ objets sont déclarés transient et donc
ne sont pas sérialisés.

Commit:
clean import, log and comments

Kumatetsu commented on MyPenelopF/src/main/java/classes/Contact.java:
```diff
@@ -34,12 +34,16 @@
 	private Dashboard dashboard;
```
du coup Dashboard no need ici

aurelienCastellarnau replied:
c'est bin vrai!

Commit:
Creation d'un controller général.
Désormais le fonctionnement est le suivant:
A la consstruction, le AppController instancie les DAO et les stock.
Lors de l'init(), le App Controller commence par initialiser les controllers requis, cette phase permet d'associer individuellement les Controllers et leur DAO respectif.
Il instancie la vue générale en lui passant les controllers requis pour les ajouter aux observers.
Une fois la vue créée, on la passe aux controllers qui pourront refresh individuellement leur zone.

Commit:
refactor pour éviter d'appeller une méthode trigger pour exécuter un display

Commit:
On donne accès au dashboardPanel à depuis une instance de la BaseFrame.

Commit:
le main se contente d'init le AppControler qui se charge du reste.
Le controller se retrouve en possession de son DAO et de sa vue et fait le lien entre les deux au travers des observers.

Kumatetsu commented on MyPenelopF/src/main/java/ihm/BaseFrame.java:
```diff
@@ -63,15 +70,13 @@ public BaseFrame() {
         
         menuPanel mPan = new menuPanel();
-        dashboardPanel dPan = new dashboardPanel();
+        dashboardPanel dPan = new dashboardPanel(this.cCtrl);
```
du veux pas qu'il l'init tout seul comme un grand dans sa classe ? il se gère tout seul comme ça...

Kumatetsu commented:
Sinon ça me semble ok

Commit:
Gros clean sur les import
deletion de fichiers .class
Rename de Message en Msg car conflit avec interface JAVA
rename et utilisation de la surcharge sur _ dans PenelopeDevLogger
éclaircissement des logs et ajout de l'appel au logger dans des fichiers clés
en gros: on log que sur les actions des controllers et les autres fichiers sont prêts
à logger du debug.
Commit sans ajout de logique, pour éclaircir la PR.

Commit:
Ajout d'un interface permettant de compléter DAO.
En interfaçant comme ça, on fige la logique des DAO. Une implémentation différente sera obligée d'implémenter ces différentes méthodes.
La logique du reste du programme est préservée dans le cas d'un nouveau DataAccessor.
Ajout de la méthode buildContactsGroups. Cette méthode remplis les groupes d'une liste de Contact.

Commit:
Ajout d'un JPanel de type GroupPanel dans le ContactPanel.
Désormais, un Contact display simplement les groups possédant son Id dans leur List de uIds.
Même si la propriété Contact.gIds existe, les méthodes utilisées actuellement passent
par groups.json et se basent sur Group.uIds pour agréger les Contact.groups.

Commit:
La BaseFrame est désormais en plein écran.
Ajout du CreateContact au mPan dans dashboardPanel.

Commit:
fin des classes Projects et Task... bonus déter...

Commit:
kill dashboard from users

Commit:
overkill.. pour le futur... ^^ oblige vraiment un développeur de l'an 2025 qui veut modifier les DAO
à implémenter tout ca comme c'est nous qu'on l'a dit ^^
